src/queue.js

```diff
@@ -204,4 +204,50 @@ Queue.prototype.shutdown = function() {
   }));
 };
 
+/**
+ * Gets queue worker count
+ */
+Queue.prototype.getWorkerCount = function() {
+  return this.workers.length;
+};
+
+/**
+ * Adds a queue worker
+ */
+Queue.prototype.addWorker = function() {
+  logger.debug('Queue: adding worker');
+  var processId = (this.specId ? this.specId + ':' : '') + this.workers.length;
+  var worker = new QueueWorker(
+    this.tasksRef,
+    processId,
+    this.sanitize,
+    this.suppressStack,
+    this.processingFunction
+  );
+  this.workers.push(worker);
+
+  if (_.isUndefined(this.specId)) {
+    worker.setTaskSpec(DEFAULT_TASK_SPEC);
+  // if the currentTaskSpec is not yet set it will be called once it's fetched
+  } else if (this.currentTaskSpec) {
+    worker.setTaskSpec(this.currentTaskSpec);
+  }
+  return worker;
+};
+
+/**
+ * Removes a queue worker if one exists
+ */
+Queue.prototype.shutdownWorker = function() {
+  logger.debug('Queue: removing worker');
+  var shutdownPromise;
+  var worker = this.workers.shift();
```

Comment:
I can see the logic there.

If you use `pop()` then you don't have to worry about the counter - the way you have things will ensure the workers always get their current index and you'll never have two workers at any point in time with the same process ID prefix. I don't think we need a counter to ensure they're unique for all time the process is running